src/calculate.rs

```diff
@@ -51,14 +52,36 @@ where
                     .signature;
                 dynamic_truncation(&digest)
             }
-            // Sha512 => unimplemented!(),
             Sha512 => return Err(Status::FunctionNotSupported),
         };
 
         Ok(truncated.to_be_bytes())
     })?
 }
 
+pub fn hmac_challenge<T>(
+    trussed: &mut T,
+    algorithm: oath::Algorithm,
+    challenge: &[u8],
+    key: &[u8],
+) -> Result<Signature>
+where
+    T: client::Client + client::HmacSha1,
+{
+    with_key(trussed, key, |trussed, key| {
+        use oath::Algorithm::*;
+        match algorithm {
+            Sha1 => {
```

Comment:
What would be needed to support SHA-2?

Reply:
An user need :-) 
Not implemented, since besides KeepassXC nothing is planned to be supported. If there are any user scenarios for that, then that's not a problem to add it.